### PR TITLE
Upload fallback keys with Crypto V2

### PR DESCRIPTION
**Pull request by Anderas**
Upload fallback keys via `keys/upload` whenever present in the body of `UploadKeysRequest`.

Additionally:
- move `MXKeysQueryScheduler` into `MXCryptoRequests` (rather than the other way around), which hides away the details of making `keys/query` requests
- rename `MXCryptoSDKLogger`
- additional log messages

### CODE REVIEW

File: MatrixSDK/Background/Crypto/MXBackgroundCryptoV2.swift [hunk 1]
```diff
@@ -60,6 +60,8 @@ class MXBackgroundCryptoV2: MXBackgroundCrypto {
           - to-device events : \(syncResponse.toDevice?.events.count ?? 0)
           - devices changed  : \(syncResponse.deviceLists?.changed?.count ?? 0)
           - devices left     : \(syncResponse.deviceLists?.left?.count ?? 0)
+          - one time keys    : \(syncResponse.deviceOneTimeKeysCount?[kMXKeySignedCurve25519Type] ?? 0)
+          - fallback keys    : \(syncResponse.unusedFallbackKeys ?? [])
```
Comment (pixlwave) [inline MatrixSDK/Background/Crypto/MXBackgroundCryptoV2.swift:64]: I'm sure its not, but double checking: Is this going to log keys that are sensitive?

**Review by pixlwave — APPROVED** [review]
Code looks good to me, although I don't have a huge amount of context around the changes :)

File: MatrixSDK/Background/Crypto/MXBackgroundCryptoV2.swift (hunk 1 above)
Comment (poljar) [reply MatrixSDK/Background/Crypto/MXBackgroundCryptoV2.swift:64]: Those keys are public, after all they get uploaded to the server.

Comment (Anderas) [reply MatrixSDK/Background/Crypto/MXBackgroundCryptoV2.swift:64]: Yea, actually in this particular log we only print the count of keys (or type in the case of fallback keys), but I have also started logging the actual keys elsewhere so your question is valid (and it is safe as we only print hte public keys)

**Pull request merged**